packages/flutter_tools/lib/src/build_runner/web_compilation_delegate.dart

```diff
@@ -195,19 +226,28 @@ class BuildRunnerWebCompilationProxy extends WebCompilationProxy {
     };
     final Status status =
         logger.startProgress('Compiling ${targets.first} for the Web...', timeout: null);
+    core.BuildResult result;
     try {
-      _builder = await BuildImpl.create(
+      result = await _runBuilder(
+        buildEnvironment,
         buildOptions,
+        targets,
+        release,
+        buildDirs,
+      );
+      return result.status == core.BuildStatus.success;
+    } on core.BuildConfigChangedException {
+      projectDirectory
```

Comment:
Ideally you wouldn't delete the entire `.dart_tool` dir - pub also uses it to cache local dependency snapshots for binaries.

You could delete `.dart_tool/build` but that would also bash over any other builds.

We do have the functionality to do a more fine grained cleanup, that only deletes the actual known outputs as well, see here https://github.com/dart-lang/build/blob/master/build_runner/lib/src/entrypoint/clean.dart#L41.

Reply:
Awesome,I tried to adapt that code to this in a way that makes sense